### PR TITLE
Feat: allow read and write mode in tree view item

### LINKED ISSUE #63: Bug: TreeViewItem blinks when node is re-redered
### Description:

`TreeViewItem` blinks when the node is re-rendered

https://github.com/powerhouse-inc/design-system/assets/20387722/27e1928a-9b97-4c6a-9a85-7e40d20a4525

### How to reproduce:

- Go to `TreeView` stories: [link here](https://design-system-git-46-treeview-create-new-041c0e-powerhouse-inc.vercel.app/?path=/story/connect-components-treeview--tree-view)
- Rename any node in the tree (3 dots options -> rename)
- Observe the results

### Additional info:

- Seems like this issue is caused by the icons and images that takes a little bit longer that the React component to render. A possible fix could be reserve the icons and images space used so when icons are rendered, there's no changes in the space used by the other elements in the component

### PR DESCRIPTION
**Pull request by ryanwolhuter**
fixes #63

the flashing bug was caused by us rendering a completely new component and icon when switching between reading and writing the tree item data. this will still happen regardless of performance and caching, because there will always be a delay when rendering a new component.

by instead keeping the same component in place and leave the rendered icon as-is, we prevent the flash and also improve the performance of the app.

### CODE REVIEW

**Comment by acaldas** [comment]
Is it possible for the selected item to keep the rounded borders? The preview is fine.
https://github.com/powerhouse-inc/design-system/assets/5501872/a36e6062-23b5-49e3-b7fb-567d7baf16a1

**Comment by ryanwolhuter** [comment]
> Is it possible for the selected item to keep the rounded borders? The preview is fine. https://github.com/powerhouse-inc/design-system/assets/5501872/a36e6062-23b5-49e3-b7fb-567d7baf16a1

Yeah can do

File: src/connect/components/tree-view/tree-view.tsx [hunk 1]
```diff
@@ -54,26 +50,20 @@ export function ConnectTreeView(props: ConnectTreeViewProps) {
     return (
         <>
             {items.map(item => {
-                if (
+                const mode =
                     item.action === ActionType.New ||
                     item.action === ActionType.Update
```
Comment (gpuente) [inline src/connect/components/tree-view/tree-view.tsx:55]: you should also consider `UpdateAndMove` and `UpdateAndCopy` here.

https://github.com/powerhouse-inc/design-system/blob/main/src/connect/components/tree-view-item/tree-view-item.tsx#L25-L30

File: src/powerhouse/components/tree-view-item/tree-view-item.tsx [hunk 2]
```diff
@@ -46,14 +49,24 @@ export const TreeViewItem: React.FC<TreeViewItemProps> = props => {
...
 
+    const inputProps = {
+        defaultValue: label,
+        onSubmitInput,
+        onCancelInput,
+        submitIcon,
+        cancelIcon,
+    };
+
+    const content = mode === 'read' ? label : <TreeViewInput {...inputProps} />;
```
Comment (gpuente) [inline src/powerhouse/components/tree-view-item/tree-view-item.tsx:60]: I think in `write` mode we should remove the `caret` and `options` icons:

Screenshot 2023-11-22 at 11 09 51

**Review by gpuente — APPROVED** [review]
LGTM, just a couple of comments

File: src/powerhouse/components/tree-view-item/tree-view-item.tsx (hunk 2 above)
Comment (ryanwolhuter) [reply src/powerhouse/components/tree-view-item/tree-view-item.tsx:60]: good idea, will do

**Pull request merged**